src/Dom/Document.php

```diff
@@ -1082,7 +1201,93 @@ private function convertAmpBindAttributes($html)
+         * DOMDocument to attempt to load it.  If the AMP HTML doesn't make use of amp-bind or similar
+         * attributes, then everything should still work.
+         *
+         * See https://github.com/ampproject/amp-wp/issues/993 for additional context on this issue.
+         * See http://php.net/manual/en/pcre.constants.php for additional info on PCRE errors.
+         */
+        return (null !== $converted) ? $converted : $html;
+    }
+
+    /**
+     * Convert AMP bind-attributes back to their original syntax.
+     *
+     * This is not guaranteed to produce the exact same result as the initial markup, as it is more of a best guess.
+     * It can end up replacing the wrong attributes if the initial markup had inconsistent styling, mixing both syntaxes
+     * for the same attribute. In either case, it will always produce working markup, so this is not that big of a deal.
+     *
+     * @see convertAmpBindAttributes() Reciprocal function.
+     * @link https://www.ampproject.org/docs/reference/components/amp-bind
+     *
+     * @param string $html HTML with amp-bind attributes converted.
+     * @return string HTML with amp-bind attributes restored.
+     */
+    public function restoreAmpBindAttributes($html)
+    {
+        if ($this->options[self::OPTION_AMP_BIND_SYNTAX] === self::AMP_BIND_SYNTAX_DATA_ATTRIBUTE) {
+            // All amp-bind attributes should remain in their converted data attribute form.
+            return $html;
+        }
+
+        if (
+            $this->options[self::OPTION_AMP_BIND_SYNTAX] === self::AMP_BIND_SYNTAX_AUTO
+            &&
+            empty($this->convertedAmpBindAttributes)
+        ) {
+            // Only previously converted amp-bind attributes should be restored, but none were converted.
+            return $html;
+        }
+
+        /**
+         * Replace callback.
+         *
+         * @param array $tagMatches Tag matches.
+         * @return string Replacement.
+         */
+        $replaceCallback = function ($tagMatches) {
+
+            // Strip the self-closing slash as long as it is not an attribute value, like for the href attribute.
+            $oldAttrs = rtrim(preg_replace('#(?<!=)/$#', '', $tagMatches['attrs']));
+
+            $newAttrs = '';
+            $offset   = 0;
+            while (preg_match(self::AMP_BIND_DATA_ATTRIBUTE_ATTR_PATTERN, substr($oldAttrs, $offset), $attrMatches)) {
+                $offset += strlen($attrMatches[0]);
+
+                $attrName = substr($attrMatches['name'], strlen(self::AMP_BIND_DATA_ATTR_PREFIX));
+                if (
+                    $this->options[self::OPTION_AMP_BIND_SYNTAX] === self::AMP_BIND_SYNTAX_SQUARE_BRACKETS
+                    ||
+                    in_array($attrName, $this->convertedAmpBindAttributes, true)
+                ) {
+                    $newAttrs .= " [{$attrName}]";
+                    if (isset($attrMatches['value'])) {
```

Comment:
Per note above, the value of an AMP-bind attribute should actually always be provided present. A bind attribute by definition has to have a value to control the state which should manage the bound attribute.

Reply:
I'm now falling back to an empty value (`=""`) instead of generating a boolean attribute.